### PR TITLE
Update CVE-2020-5957.json

### PR DESCRIPTION
**Pull request by PSIRT-NVIDIA**
NVIDIA CVE submission

### CODE REVIEW

File: 2020/5xxx/CVE-2020-5957.json [hunk 1]
```diff
@@ -1,18 +1,60 @@
 {
```
Comment (cve-team) [inline 2020/5xxx/CVE-2020-5957.json]: 'psirt@nvidia.com\r\n' does not match '^([a-zA-Z0-9_\\-\\.]+)@([a-zA-Z0-9_\\-\\.]+)\\.([a-zA-Z]{2,5})$'

File: 2020/5xxx/CVE-2020-5957.json [hunk 2]
```diff
@@ -1,18 +1,60 @@
...
+				{
+					"product" : {
+						"product_data" : [
+							{
+								"product_name" : "NVIDIA GPU Display Driver\r\n",
+								"version" : {
+									"version_data" : [
+										{
+											"version_value" : "ALL\r\n"
+										}
```
Comment (iamleot) [inline 2020/5xxx/CVE-2020-5957.json]: Can you please adjust it to reflect versions mentioned in the reference? (it seems that not all versions are affected, i.e. there are fixed versions)

File: 2020/5xxx/CVE-2020-5957.json [hunk 3]
```diff
@@ -1,18 +1,60 @@
...
+		}
+	},
+	"data_format" : "MITRE",
+	"data_type" : "CVE",
+	"data_version" : "4.0",
+	"description" : {
+		"description_data" : [
+			{
+				"lang" : "eng",
+				"value" : "NVIDIA Windows GPU Display Driver, all versions, contains a vulnerability in the NVIDIA Control Panel component in which an attacker with local system access can corrupt a system file, which may lead to denial of service or escalation of privileges. "
```
Comment (iamleot) [inline 2020/5xxx/CVE-2020-5957.json]: Can you please substitute `all versions` with the version including security updates as listed in the reference?

**Comment by PSIRT-NVIDIA** [comment]
I submitted the changes with the \r\n characters .  
the other suggestions is not applicable in our descriptions

**Comment by iamleot** [comment]
Can you please elaborate further why?

**Comment by PSIRT-NVIDIA** [comment]
I understand your point, but we have many branches and version affected and doesn't make sense to put all of them in the description because it makes it too long and not easy to ready. When we release new updates, we update our security bulletin

**Comment by iamleot** [comment]
At least as final CVE "consumer", directly putting the affected versions there it's much more easy to follow and consistent with the information in the security bulletin.

**Comment by PSIRT-NVIDIA** [comment]
We have discussed this with MITRE in the past on this, and this is format we have agreed upon which is the best option.   (see example: CVE‑2019‑5690)

**Comment by iamleot** [comment]
I think it would be better to not omit possible information in the CVE, making it as possible self-consistent and use references for further details as a final "consumer"/reader of the CVE.  The "best" is probably subjective here.

Also, directly quoting <https://cve.mitre.org/about/faqs.html#cve_entry_descriptions_created> (emphasis is mine):

> Descriptions should be unique and provide the relevant details to help users (1) find the CVE Entry for a specific vulnerability, and/or (2) distinguish between similar-looking vulnerabilities.
>
> Ideally, Descriptions include details such as the name of the affected product and vendor, a **summary of affected versions**, the vulnerability type, the impact, the access that an attacker requires to exploit the vulnerability, and the important code components or inputs that are involved.
> 
> However, if this information is not available to the submitter of a CVE Entry, not all Descriptions will include all these details. For example, to write a CVE Description, the CVE Team analyzes public, third-party reports of vulnerabilities (i.e., "references"); extracts the relevant information from each reference; resolves any conflicting information or inconsistent usage of terminology; and then writes the description. In this example, the CVE Team only has access to publicly available information which may not include all the details of the ideal Descriptions.

I am completely unaffiliated to the MITRE but I think that adding all important information - like versions that are affected when that it is a known information - to the CVE make them better and the community referencing CVE benefits of that.


Thanks!

**Comment by PSIRT-NVIDIA** [comment]
Hi,

I am aware of the MITRE page for CVE. To understand this, For GPU Display Driver, we have 6 different products and each of them can have 5 affected branches and each of them have unique driver versions. If might see our SB you will notice how complicated it can get. That is the reason we have created tables and format that is relatively easy to relate and mappings. We always have the SB reference attached when we post the CVE. 

We have explored all the options and that is why we have put in table formats in our Security Bulletin.  For our other products like SHIELD and L4T, we do follow the format and add affected versions is doable.

For our other products that we only have one products and 2-3 versions affected we follow the normal guidelines, but for GPU Display Driver CVEs we have to make it generic in the title.

**Pull request merged**